tensorflow_addons/optimizers/weight_decay_optimizers.py

```diff
@@ -226,6 +238,25 @@ def _resource_apply_sparse(self, grad, var, indices, apply_state=None):
                 grad, var, indices, apply_state=apply_state
             )
 
+    def _do_use_weight_decay(self, var):
+        """Whether to use L2 weight decay for `var`."""
+        if not self._decay_var_list or var.ref() in self._decay_var_list:
+            if self.exclude_from_weight_decay:
+                var_name = self._get_variable_name(var.name)
+                for r in self.exclude_from_weight_decay:
+                    if re.search(r, var_name) is not None:
+                        # print("Filtered:", var_name)
+                        return False
+            return True
+        return False
+
+    def _get_variable_name(self, param_name):
```

Comment:
Do we need this or wouldn't re.search find the substring anyway?

Reply:
I'm not sure about this, as this is from `lamb.py`. I kept a function `get_variable_name` in `optimizers/utils.py` but not using it in `is_variable_excluded_by_regexes`. Maybe some consideration from the author? @junjiek